examples/routing_service/routing_service_msql_adapter/cpp/rs_dll/RsMsqlShapesAdapter_w_types.xml

```diff
@@ -0,0 +1,385 @@
+      </enum>
+      <struct name= "ShapeType">
+        <member name="color" stringMaxLength="128" type="string" key="true"/>
+        <member name="x" type="int32"/>
+        <member name="y" type="int32"/>
+        <member name="shapesize" type="int32"/>
+      </struct>
+      <struct name= "ShapeTypeExtended" baseType="ShapeType">
+        <member name="fillKind" type="nonBasic"  nonBasicTypeName= "ShapeFillKind"/>
+        <member name="angle" type="float32"/>
+      </struct>
+    </module>
+  </types>
+
+  <routing_service name="RS_DDS_To_MSQL">
+
+    <annotation>
+      <documentation>
+        Read DDS shapes on domain 0 and write to MSQL database
+      </documentation>
+    </annotation>
+
+    <administration>
+      <domain_id>0</domain_id>
+      <distributed_logger>
+        <enabled>true</enabled>
+      </distributed_logger>
+    </administration>
+
+    <domain_route name="DDS_to_MSQL_Route">
+      <participant name="DDS_Participant">
+        <domain_id>0</domain_id>
+        <participant_qos base_name="RsMsqlQosLib::RsMsqlQosReliableProfile"/>
+        <!-- need this since types defined in XML, instead of using discovery -->
+        <registered_type name="ShapeType" type_name="XMLTypes::ShapeTypeExtended"/>
+      </participant>
+
+      <!-- Configures a custom, adapter-based connection -->
+      <connection name="msql_connection" plugin_name="MsqlDataPlugins::MsqlAdapter">
+        <!-- Add properties here to configure the Connection instance.-->
+      </connection>
+
+      <session name="dds_to_msql_session" enabled="true">
+
+        <route name="Squares Route">
+          <!-- To use types discovered in either one of the domains for the creation of both  
+          the StreamReader and StreamWriter, you must set the <route_types> tag to true. -->
+          <route_types>true</route_types>
+          <dds_input name="Squares" participant="DDS_Participant">
+            <creation_mode>ON_ROUTE_MATCH</creation_mode>
+            <datareader_qos base_name="RsMsqlQosLib::RsMsqlQosReliableProfile"/>
+            <topic_name>Square</topic_name>
+            <registered_type_name>ShapeType</registered_type_name>
+          </dds_input>
+          <output connection="msql_connection" name="MsqlSquares">
+            <creation_mode>ON_ROUTE_MATCH</creation_mode>
+            <stream_name>Square</stream_name>
+            <registered_type_name>ShapeType</registered_type_name>
+            <!-- Add properties here to configure the StreamWriter. -->
+            <property>
+              <value>
+                <element>
+                  <name>msql.db.servername</name>
+                  <value>RTI-10591\SQLEXPRESS</value>
+                </element>
+                <element>
+                  <name>msql.db.dbname</name>
+                  <value>Shapes</value>
+                </element>
+                <element>
+                  <name>msql.db.tablename</name>
+                  <value>Shapes_table</value>
```

Comment:
The table we're referring to here, does it have to exist prior to running the example? That's what I gathered from the code, but I may be missing something.
In that case, it is expected that the user is going to create it, right?

Reply:
In that case I would strongly suggest you add a section in the comments with how to set up the database (I may have read over it, or I may have not got to that point yet - please ignore if that is the case). This would include general MSQL setup for the example and SQL `CREATE TABLE` statement to apply.

Reply:
This example is associated with a blog post. The blog clearly states that the user must have MSQL installed and be able to connect to the database. The blog describes how to create the table. https://www.rti.com/blog/msql-database-routing-service-adapter?fbclid=IwAR2QVrZAQtE5pl94CrPYJLZ3p24ipKs7EdxvUQiZkuuDI_aBlzpn3Wy8_jE. I placed a link to the blog in the toplevel Readme.md file.